### DISCUSSION TITLE
Add QPIGS2 support for dual mppt inverter

### DISCUSSION THREAD

**Discussion by johnjohn031** [lavron/docker-voltronic-mqtt#5] [General]
Hi,
I've got some inverter AXPERT MAX TWIN 11kw with dual mppt tracker and value are read with QPIGS2 command.

Can you add this command?

Thanks,
John
[Axpert MAXII,VMIV, MKSIV Communication Protocol20221104-Twin(1).pdf](https://github.com/user-attachments/files/17216693/Axpert.MAXII.VMIV.MKSIV.Communication.Protocol20221104-Twin.1.pdf),

**Comment by johnjohn031** [comment]
Hi,
I've modified the loop         
```
        inverter.update("QMOD")
        inverter.update("QPIGS")
        inverter.update("QPIGS2")
        inverter.update("QPIWS")
        inverter.update("QLT")
        inverter.update("QET")
        inverter.update("QT")
```

and modified/added some field.
```
{
    "QMOD": {
        "mode": {"index": 0, "type": "str"}
    },
    "QPIGS": {
        "grid_voltage": {"index": 0, "type": "float"},
        "grid_frequency": {"index": 1, "type": "float"},
        "AC_Output_voltage": {"index": 2, "type": "float"},
        "AC_Output_frequency": {"index": 3, "type": "float"},
        "AC_Output_Apparent_Power": {"index": 4, "type": "int"},
        "output_load_watt": {"index": 5, "type": "int"},
        "Output_load_percent": {"index": 6, "type": "int"},
        "BUS_Voltage": {"index": 7, "type": "int"},
        "Battery_Voltage": {"index": 8, "type": "float"},
        "Battery_Voltage_from_SCC": {"index": 14, "type": "float"},
        "Battery_charging_current": {"index": 9, "type": "int"},
        "Battery_discharge_current": {"index": 15, "type": "int"},
        "battery_capacity": {"index": 10, "type": "int"},
        "heatsink_temperature": {"index": 11, "type": "int"},
        "pv_input_voltage": {"index": 13, "type": "float"},
        "pv_input_current": {"index": 12, "type": "float"},
        "pv_input_watt": {"index": 19, "type": "int"}
    },
    "QPIGS2": {
        "pv2_input_voltage": {"index": 1, "type": "float"},
        "pv2_input_current": {"index": 0, "type": "float"},
        "pv2_input_watt": {"index": 2, "type": "int"}
    },
    "QPIWS": {
        "warnings": {"index": 0, "type": "str"}
    },
    "QLT": {
        "Total_output_load_energy": {"index": 0, "type": "int"}
    },
    "QET": {
        "Total_PV_generated_energy": {"index": 0, "type": "int"}
    },
    "QT": {
        "Last_Inverter_read": {"index": 0, "type": "int"}
    }
}
```

It work fine.

**Comment by johnjohn031** [comment]
I've used the MQTT_TOPIC_PREFIX option that is not mentioned on the readme and work fine.
I've 2 dockers, each on an inverter and each with a different prefix.

**Comment by lavron** [comment]
@johnjohn031 Great work, thanks for sharing!

**Reply by ptest8794** [reply]
> @johnjohn031 Great work, thanks for sharing!

Hello @lavron ,

I was wondering if it was possible to update the Docker src file to add more visibility with @johnjohn031  change.
Thanks a lot.

**Comment by eposza** [comment]
Could we get more info on the configuration, what to use there is no documentation but should be very usefull

**Comment by 1967luca** [comment]
Buongiorno sono riuscito a far funzionare tutto, mi chiedo perche io non riesco ad arrivarci come sia possibile avere la produzione giornaliera da fotovoltaico come da manuale che allego .
Ringrazio se qualcuno con più competenze voglia dare un contributo.



QEYyyyy<CRC><cr>: Query PV generated energy of year
Computer: QEYyyyy<cr>
Device: (NNNNNNNN<CRC><cr>
yyyy: Year, y: 0~9
NNNNNNNN: Generated energy, N: 0~9, unit: Wh
2.25 QEMyyyymm<CRC><cr>: Query PV generated energy of month
Computer: QEMyyyymm <CRC><cr>
Device: (NNNNNNNN<CRC><cr>
yyyy: Year, y: 0~9
mm: Month, m: 0~9
NNNNNNNN: Generated energy, N: 0~9, unit: Wh
2.26 QEDyyyymmdd<CRC><cr>: Query PV generated energy of day
Computer: QEDyyyymmdd<CRC><cr>
Device: (NNNNNNNN<CRC><cr>
yyyy: Year, y: 0~9
mm: Month, m: 0~9
dd: Day, d: 0~9
NNNNNNNN: Generated energy, N: 0~9, unit: Wh

**Reply by lavron** [reply]
Partly it's solved here:
https://github.com/lavron/docker-voltronic-mqtt/discussions/8
Minor changes would be needed in Voltronic.unpack_data(), and that's basically all.
@1967luca will you handle it? I could implement it in the next release, but can't promise when that would be.
